xun/functions/util.py

```diff
@@ -203,6 +203,51 @@ def assignment_target_names(t):
         raise TypeError('Unsupported target {}'.format(type(t)))
 
 
+def assignment_target_shape(t):
+    """Assignment target shape
+
+    Given a target expression node, of type ast.Tuple or ast.List,
+    return the shape of the target.
+
+    Examples
+    --------
+    >>> target = 'a, (b, c)'
+    >>> ast_target = ast.parse(target).body[0].value
+    >>> assignment_target_shape(ast_target)
+    (1, (2,))
+
+    """
+    assert isinstance(t, (ast.Tuple, ast.List))
+
+    inner_tuple = ()
+    count_Names = 0
+    n_elements = len(t.elts)
+    for el in t.elts:
+        if isinstance(el, ast.Name):
+            count_Names += 1
+            inner_tuple += (1, )
+        else:
+            if isinstance(el, (ast.Tuple, ast.List)):
+                inner_tuple += (assignment_target_shape(el), )
+            elif isinstance(el, ast.Starred):
+                inner_tuple += (0, )
+            else:
+                raise TypeError("Invalid node in target tuple")
+
+    if count_Names == n_elements:
+        inner_tuple = (n_elements, )
+
+    return inner_tuple
+
+
+def shape_to_ast_tuple(shape):
+    """
+    Given a tuple shape, return an ast representation of that tuple
+    """
+    assert isinstance(shape, tuple)
+    return ast.parse(str(shape)).body[0].value
```

Comment:
Maybe, probably not, perhaps a security hole. Example:

```python
class A:
    def __repr__(self):
        return 'os.rm(\'/\')'
shape = (1, (2, A()))
```

Let's change it to a simple recursive call, or loop

Reply:
Changed it to a recursive call.